validator/project_validator_test.go

```diff
@@ -1032,6 +1032,21 @@ func TestValidatePluginCommands(t *testing.T) {
 			So(validatePluginCommands(project), ShouldNotResemble, ValidationErrors{})
 			So(len(validatePluginCommands(project)), ShouldEqual, 1)
 		})
+		Convey("an error should be thrown if a function plugin command doesn't have commands", func() {
+			project := &model.Project{
+				Functions: map[string]*model.YAMLCommandSet{
+					"funcOne": {
+						SingleCommand: &model.PluginCommandConf{
```

Comment:
I'm not sure I follow. Doesn't this function _have_ a command?

Reply:
There's some ambiguity in terminology - the ticket is referring to the command field (ie. shell.exec) being blank. The data structure here is also called a command which contains the command itself and metadata

Reply:
it seems like we only added "is the command list empty" validation, and not "is the command misspecified.?" 

Is it possible to add something that attempts to parse the command? by calling command.Render() on the command? My gut is that this is a cycle, but4 could also be a problem

Reply:
The command should already be parsed and the existence of the command should be checked in `validateCommands` when it calls `Render()`: https://github.com/evergreen-ci/evergreen/blob/65e7945eed4d707198982cc71890b7e1e12b43f1/validator/project_validator.go#L708